pkg/templates/templates_test.go

```diff
@@ -56,10 +57,66 @@ func TestMain(m *testing.M) {
 	}
 	simpleClient.CoreV1().ConfigMaps(testns).Create(context.TODO(), &configmap, metav1.CreateOptions{})
 
-	InitializeKubeClient(&simpleClient, nil)
+	resolver, err := New(&simpleClient, c)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return resolver
+}
+
+func TestNew(t *testing.T) {
+	t.Parallel()
+	var simpleClient kubernetes.Interface = fake.NewSimpleClientset()
+
+	testcases := []struct {
+		testName string
+		config   Config
+	}{
+		{"KubeConfig set", Config{KubeConfig: &rest.Config{}}},
+		{"KubeAPIResourceList set", Config{KubeAPIResourceList: []*metav1.APIResourceList{}}},
+	}
+
+	for _, test := range testcases {
+		test := test
+		t.Run(test.testName, func(t *testing.T) {
+			t.Parallel()
+			_, err := New(&simpleClient, test.config)
+			if err != nil {
+				t.Fatalf("No error was expected: %v", err)
+			}
+		})
+	}
+}
 
-	exitVal := m.Run()
-	os.Exit(exitVal)
+func TestNewFailures(t *testing.T) {
+	t.Parallel()
+	var simpleClient kubernetes.Interface = fake.NewSimpleClientset()
+
+	testcases := []struct {
+		kubeClient  *kubernetes.Interface
+		config      Config
+		expectedErr string
+	}{
+		{nil, Config{}, "kubeClient must be a non-nil value"},
+		{&simpleClient, Config{}, "the configuration must have either KubeAPIResourceList or kubeConfig set"},
+	}
+
+	for _, test := range testcases {
+		testName := fmt.Sprintf("expectedErr=%s", test.expectedErr)
+		test := test
```

Comment:
Since each test run is parallel, we need to redeclare `test` so that each parallel test gets the correct value for `test` and not what `test` happens to be in the range loop at the time of test execution.

The go linting on this repo actually catches this which is nice.